packages/library-base/src/commonMain/kotlin/io/realm/kotlin/internal/KeyPathFlowable.kt

```diff
@@ -21,7 +21,17 @@ import kotlinx.coroutines.flow.Flow
 /**
  * A __flowable__ is an internal entity that supports listening to changes on the type [T] as a
  * [Flow].
+ *
+ * This comes in two variants: One variant for flows on things like Objects, Results and Lists that
+ * allow you to pass in keypaths to restrict notifications to certain properties and one variant
+ * for flows on entities that doesn't support this, e.g. flows on query aggregates or the Realm as
+ * whole.
  */
+
+internal interface KeyPathFlowable<T> {
+    fun asFlow(keyPaths: List<String>? = null): Flow<T>
+}
+
 internal interface Flowable<T> {
```

Comment:
If we don't really have a connection of 1-1 connection between `Flowable` to `FlowableTests` then I guess `Flowable` could just go since only implemented by `RealmImpl` now, right?

Reply:
Yes, good point.